tiling.js

```diff
@@ -1515,13 +1522,164 @@ class Spaces extends Map {
                 continue;
             monitor.clickOverlay.activate();
         }
+
+        inPreviewMode = {stack: false, sequence: false};
+    }
+
+    _initWorkspaceSequence() {
+
+        if (inPreview) {
+            return;
+        }
+
+        // Always show the topbar when using the workspace stack
+        inPreview = true;
+        inPreviewMode = {stack: false, sequence: true};
+
+        TopBar.show();
+
+        let currentSpace = this.spaceOf(workspaceManager.get_active_workspace());
+        let monitorSpaces = [];
+
+        for (let [workspace, space] of this) {
+            if (space.monitor === currentSpace.monitor) {
+                monitorSpaces.push(space)
+            }
+        }
+
+        if (Main.panel.statusArea.appMenu) {
+            Main.panel.statusArea.appMenu.container.hide();
+        }
+
+        let currentMonitor = currentSpace.monitor;
+        this.selectedSpace = currentSpace;
+
+        let cloneParent = currentSpace.clip.get_parent();
+
+        const scale = 0.8;
```

Comment:
should that also be true in _initWorkspaceStack

Reply:
That would probably make the stack animation far too busy, as we'd see all the spaces being scaled down at the same time.

Reply:
let me rephrase. _initWorkspaceSequence has a `scale=1` in this last revision but _initWorkspaceStack has scale of `0.9` ... so what I am wondering if _initWorkspaceStack scale should match _initWorkspaceSequence. (sorry if I am confused)

Reply:
So `_initWorkspaceStack` (and `_initWorkspaceSequence`) is there to to set up the visual state before animations takes place, ie. position and potentially scale the non-visible spaces. So they serve the same function, but that doesn't mean they necessarily will share the same code.

For the workspace stack that means we want to position and scale the hidden spaces when the active space covers the the whole screen.

For sequential navigation we want to layout the hidden spaces in order «around» the active space. In this case we don't want the spaces to overlap at any time, partly because it looks better, but also to avoid having to manually manage the stacking (z-axis) order of the spaces. The easiest way to do this as far as I can see is having the spaces start out at `scale=1` and then scale down the sequence of spaces together.

So I'd say we should just leave the workspace stack alone for now. We might want to try to match the «active» scaling of the stack and sequence for visual consistency, but we can tackle that later (eg. the stack scaling is eg. tuned to always have overlap).

Reply:
So I looked over animateToSpace and switchToWorkspace and from what I see _initWorkspaceSequence is called before animateToSpace occurs and from what I can tell it's sufficient and have not had any side effects I can see after testing. I did put a conditional around some code in animateToSpace specific to stack arrangement but that was minor. please see 903dea3

also I removed a tween from _initWorkspaceSequence it seemed unneed to me (I could be wrong) but _initWorkspaceSequence also does not have it. Not sure what use case you were thinking of there, but in my testing it made no difference. See cfef02e